client/domain/PreRequisiteEvalResultObjectFactory.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of
+ * PreRequisiteCheckResult domain objects.
+ */
+
+tie.factory('PreRequisiteEvalResultObjectFactory', [
+  function() {
+    var PreRequisiteEvalResult = function(
+      code, preReqErrorMessage, starterCode) {
+      this._code = code;
+      this._preReqErrorMessage = preReqErrorMessage;
+      this._starterCode = starterCode;
+    };
+
+    // Instance methods
+    PreRequisiteEvalResult.prototype.getCode = function() {
```

Comment:
Ah ok -- I checked SolutionHandlerService, and the codeEvalResult is created after running the preprocessed code, so it would be different from the code stored in the prereq check result object (if we had indeed stored code in it). Since we didn't, though, that's a moot point, and I think what is currently in master looks good. We can always add the raw code later to the prereq result if we need it.

Given this, I don't think we need to pull the code out of CodeEvalResult, either now or in the future. Would you agree?